models/DataObject/Concrete.php

```diff
@@ -713,7 +713,10 @@ public function __wakeup(): void
             // We're reloading version data, there might be fields that now implement the ObjectAwareFieldInterface but
             // aren't included in the $this->__objectAwareFields array - for example versions created in Pimcore <= 10.x
             // containing LocalizedFields. Verify all fields in this object.
-            foreach (get_object_vars($this) as $propertyValue) {
+            foreach (get_object_vars($this) as $propertyKey => $propertyValue) {
+                if (str_ends_with($propertyKey, 'o_published')) {
+                    $this->published = (bool) $propertyValue;
+                }
```

Comment:
to address the input that it should be covering all the keys, what about something like:

```suggestion
                $bcPropertyKey = DataObject\Service::getVersionDependentDatabaseColumnName($propertyKey);
                if (propertyKey !== bcPropertyKey){
                    $this->{$bcPropertyKey} = $propertyValue;
                }
```

Reply:
Mhhh, but it doesn't address the protected issue that is prefixed with a special char..